projects/common/Math/Vector2.h

```diff
@@ -34,22 +34,24 @@ class Vector2
 public:
 	// Constructors
 	Vector2() {};
-	Vector2(f32 inX, f32 inY) { x = inX; y = inY; }
-	Vector2(f32 in) { x = in; y = in; }
-	Vector2(u8 inX, u8 inY) { x = static_cast<f32>(inX); y = static_cast<f32>(inY); }
-	Vector2(u8 in) { x = static_cast<f32>(in); y = static_cast<f32>(in); }
-	Vector2(u16 inX, u16 inY) { x = static_cast<f32>(inX); y = static_cast<f32>(inY); }
-	Vector2(u16 in) { x = static_cast<f32>(in); y = static_cast<f32>(in); }
-	Vector2(u32 inX, u32 inY) { x = static_cast<f32>(inX); y = static_cast<f32>(inY); }
-	Vector2(u32 in) { x = static_cast<f32>(in); y = static_cast<f32>(in); }
-	Vector2(i8 inX, i8 inY) { x = static_cast<f32>(inX); y = static_cast<f32>(inY); }
-	Vector2(i8 in) { x = static_cast<f32>(in); y = static_cast<f32>(in); }
-	Vector2(i16 inX, i16 inY) { x = static_cast<f32>(inX); y = static_cast<f32>(inY); }
-	Vector2(i16 in) { x = static_cast<f32>(in); y = static_cast<f32>(in); }
-	Vector2(i32 inX, i32 inY) { x = static_cast<f32>(inX); y = static_cast<f32>(inY); }
-	Vector2(i32 in) { x = static_cast<f32>(in); y = static_cast<f32>(in); }
-
-	Vector2(Vector2& in) { x = in.x; y = in.y; }
+	Vector2(float x, float y) : x(x), y(y) {}
```

Comment:
Clang seemed to get very confused by all of these different constructors and would refuse to compile. I've just commented out the ones that are not used for now and it solves the problem but this one might require further research.

Reply:
I will have to get my Clang running and look into fixing this, but I'm not sure why this would happen and I don't know if I have an opportunity to look at it until worst case Thursday.